pkg/sql/create_type.go

```diff
@@ -222,6 +218,18 @@ func CreateEnumArrayTypeDesc(
 	switch t := typDesc.Kind; t {
 	case descpb.TypeDescriptor_ENUM, descpb.TypeDescriptor_MULTIREGION_ENUM:
 		elemTyp = types.MakeEnum(catid.TypeIDToOID(typDesc.GetID()), catid.TypeIDToOID(id))
+	case descpb.TypeDescriptor_COMPOSITE:
+		for _, t := range typDesc.Composite.TupleContents {
+			if t.UserDefined() {
+				return nil, unimplemented.NewWithIssue(91779,
+					"composite types that reference user-defined types not yet supported")
+			}
+			if t.TypeMeta.ImplicitRecordType {
+				return nil, unimplemented.NewWithIssue(70099,
+					"cannot use table record type as part of composite type")
+			}
+		}
```

Comment:
I see these checks when creating the array type descriptor but can't find them when creating the composite-type descriptor, is this expected?